lib/request-handler.js

```diff
@@ -0,0 +1,148 @@
+"use strict";
+
+
+/**
+ * @typedef {Object} requestHandler
+ * @function buildRequestOptions
+ * @function checkResponse
+ * @function createUrl
+ */
+
+
+/**
+ * Creates requestHandler.
+ *
+ * @param {bluebird} bluebird
+ * @param {dateService} dateService
+ * @param {string} host
+ * @param {logger} logger
+ * @param {urijs} Uri
+ * @return {requestHandler}
+ */
+module.exports = (bluebird, dateService, host, logger, Uri) => {
+    // This error object is a last ditch attempt to supply info if
+    // OpenToken does not provide a body.
+    const error = {
+        401: "unauthorized",
+        403: "forbidden",
+        404: "resource_not_found",
+        405: "method_not_allowed",
+        500: "internal_server_error",
+        unknown: "unknown"
+    };
+
+
+    /**
+     * Build request options for a request to OpenToken.
+     *
+     * @param {string} url
+     * @param {string} method
+     * @param {string} contentType
+     * @return {Object}
+     */
+    function buildRequestOptions(url, method, contentType) {
+        if (typeof contentType === "undefined") {
+            contentType = "text/plain";
+        }
+
+        return {
+            headers: {
+                "content-type": contentType,
+                "x-opentoken-date": dateService.now(),
+                host
+            },
+            method,
+            url
+        };
+    }
+
+
+    /**
+     * Creates error from an error response from OpenToken.
+     *
+     * @param {string} body
+     * @param {Number} statusCode
+     * @throw {Error}
+     */
+    function createErrorResponse(body, statusCode) {
+        var message;
+
+        // If OpenToken does not return a message in the body, I attempt to map
+        // the status code to an error. If all else fails, I return an unknown error.
+        try {
+            body = JSON.parse(body);
+        } catch (e) {
+            logger.debug("Failed to parse response body to JSON.");
+        }
+
+        if (body) {
+            message = body.message;
+        }
+
+        if (!message) {
+            if (Object.keys(error).indexOf(statusCode.toString()) >= 0) {
```

Comment:
Was thinking about this.  I suppose my alternative would break if someone modified `Object.prototype.whatever = true` and the statuscode matched `whatever`, but both of our things would break.  The truly correct one would be

    if (error.hasOwnProperty(statusCode)) {

But that is not what I normally see.  Besides, how often do you see people modifying the `Object.prototype` to include properties that are numbers?  :-)